edb/schema/schema.py

```diff
@@ -1413,7 +1420,11 @@ def getter(schema: FlatSchema, name: sn.Name) -> Optional[so.Object]:
         if obj is not so.NoDefault:
             # We do our own type check, instead of using get_by_id's, so
             # we can produce a user-facing error message.
-            if obj and type is not None and not isinstance(obj, type):
+            if (
+                obj
+                and type is not None
+                and (not skip_type_check and not isinstance(obj, type))
+            ):
```

Comment:
We should not do this. It makes the type annotations wrong for the overloads.

It would be OK if we treated it as not being found if the type is wrong and `skip_type_check` is set, but I think that's probably not worth the hassle, and you should get rid of `skip_type_check` and handle it at the call site.

You could either pass `Object` as the type and do all the checks there, or you could catch the exception.
Probably the first is cleaner.